src/components/ChainSelectionPopover/ChainSelectionPopover.tsx

```diff
@@ -0,0 +1,121 @@
+import { useRef, RefObject } from "react";
+import { useTranslation } from "react-i18next";
+
+import { ChainIds } from "@airswap/constants";
+
+import { useAppDispatch } from "../../app/hooks";
+import nativeCurrency from "../../constants/nativeCurrency";
+import { CHAIN_PARAMS } from "../../constants/supportedNetworks";
+import { setWalletConnected } from "../../features/wallet/walletSlice";
+import {
+  Container,
+  NetworksContainer,
+  NetworkButton,
+  NetworkIcon,
+} from "./ChainSelectionPopover.styles";
+import PopoverSection from "./subcomponents/PopoverSection/PopoverSection";
+
+type ChainSelectionPopoverPropsType = {
+  chainId: number | undefined;
+  account: string | undefined | null;
```

Comment:
When you're not connected to a network just don't render the button and popover.

```suggestion
  chainId: number;
  account: string;
```